src/container/amlManager.ts

```diff
@@ -0,0 +1,154 @@
+        const imageItem: vscode.QuickPickItem = await vscode.window.showQuickPick(this.loadAmlImageItems(workspace, session),
+            { placeHolder: "Select Image", ignoreFocusOut: true });
+        return imageItem;
+    }
+
+    private async loadAmlImageItems(workspace: Workspace, session: AzureSession): Promise<vscode.QuickPickItem[]> {
+        try {
+            const modelMgmtEndpoint: string = await this.getModelMgmtEndpoint(workspace);
+            const credentials: ServiceClientCredentials = session.credentials;
+            const client: ServiceClient = new ServiceClient(credentials);
+            const result: HttpOperationResponse<{ value: IImage[] }> = await client.sendRequestWithHttpOperationResponse<{ value: IImage[] }>({
+                method: "GET",
+                baseUrl: modelMgmtEndpoint,
+                pathTemplate: `/api${workspace.id}/images`,
+                queryParameters: { "api-version": Constants.amlApiVersion },
+                serializationMapper: undefined,
+                deserializationMapper: undefined,
+            });
+
+            return result.body.value.map((image: IImage) => {
+                return { label: image.name, description: image.imageLocation };
+            });
+        } catch (error) {
+            error.message = `Error fetching image list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private getWorkspaceNameCounts(workspaces: Workspace[]): Map<string, number> {
+        const counts: Map<string, number> = new Map<string, number>();
+        workspaces.forEach((workspace: Workspace) => {
+            counts.set(workspace.name, counts.has(workspace.name) ? counts.get(workspace.name) + 1 : 1);
+        });
+
+        return counts;
+    }
+
+    private async getModelMgmtEndpoint(workspace: Workspace): Promise<string> {
+        const historyEndpoint: string = await this.getHistoryEndpoint(workspace);
+        let location: string;
+        const res: string[] = historyEndpoint.match(/\/\/(.*?)\./);
+        if (res.length < 2) {
+            throw new Error("No available endpoint");
+        } else {
+            location = res[1];
+        }
+
+        return `https://${location}.modelmanagement.azureml.net`;
+    }
+
+    private async getHistoryEndpoint(workspace: Workspace): Promise<string> {
+        const client: ServiceClient = new ServiceClient();
+        const result: HttpOperationResponse<IEndPoints> = await client.sendRequestWithHttpOperationResponse<IEndPoints>({
+            method: "GET",
+            url: workspace.discoveryUrl,
+            serializationMapper: undefined,
+            deserializationMapper: undefined,
+        });
+        if (result.response.statusCode === 200) {
+            return result.body.history;
+        } else {
+            throw new Error("API end points not found."); // TODO: return defaults by region
```

Comment:
Do we need to show detailed error messages such as 5xx  errors

Reply:
This code snippet is copied from AI tools. Given we have little documentation about the API, I tend to keep it as is.